src/library/scala/sys/process/ProcessImpl.scala

```diff
@@ -32,14 +35,14 @@ private[process] trait ProcessImpl {
   }
   private[process] object Future {
     def apply[T](f: => T): (Thread, () => T) = {
-      val result = new SyncVar[Either[Throwable, T]]
+      val result = Promise[Either[Throwable, T]]()
       def run(): Unit =
-        try result.put(Right(f))
-        catch { case e: Exception => result.put(Left(e)) }
+        try result.trySuccess(Right(f))
+        catch { case e: Throwable => result.trySuccess(Left(e)) }
```

Comment:
Not 100% sure about this, since this will suppress fatal errors.

```suggestion
        catch { case NonFatal(t) => result.trySuccess(Left(t)) }
```